core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -0,0 +1,444 @@
+                      protected val terraBucketWriterRole: String
+)(implicit protected val executionContext: ExecutionContext, val openTelemetry: OpenTelemetryMetrics[IO])
+    extends LazyLogging {
+
+  private val openTelemetryTags: Map[String, String] = Map("service" -> "FastPassService")
+  private def samWorkspaceRoleToGoogleProjectIamRoles(samResourceRole: SamResourceRole) =
+    samResourceRole match {
+      case SamWorkspaceRoles.projectOwner =>
+        Set(terraBillingProjectOwnerRole, terraWorkspaceCanComputeRole, terraWorkspaceNextflowRole)
+      case SamWorkspaceRoles.owner      => Set(terraWorkspaceCanComputeRole, terraWorkspaceNextflowRole)
+      case SamWorkspaceRoles.canCompute => Set(terraWorkspaceCanComputeRole, terraWorkspaceNextflowRole)
+      case _                            => Set.empty[String]
+    }
+
+  private def samWorkspaceRolesToGoogleBucketIamRoles(samResourceRole: SamResourceRole) =
+    samResourceRole match {
+      case SamWorkspaceRoles.projectOwner => Set(terraBucketWriterRole)
+      case SamWorkspaceRoles.owner        => Set(terraBucketWriterRole)
+      case SamWorkspaceRoles.writer       => Set(terraBucketWriterRole)
+      case SamWorkspaceRoles.reader       => Set(terraBucketReaderRole)
+      case _                              => Set.empty[String]
+    }
+
+  def setupFastPassForUserInNewWorkspace(workspace: Workspace): ReadWriteAction[Unit] = {
+    if (!config.enabled) {
+      logger.debug(s"FastPass is disabled. Will not grant FastPass access to ${workspace.toWorkspaceName}")
+      return DBIO.successful()
+    }
+
+    DBIO.from(quotaAvailableForNewWorkspaceFastPassGrants(workspace)).flatMap { quotaAvailable =>
+      if (quotaAvailable) {
+        logger
+          .info(s"Adding FastPass access for ${ctx.userInfo.userEmail.value} in workspace ${workspace.toWorkspaceName}")
+        val expirationDate = DateTime.now(DateTimeZone.UTC).plus(config.grantPeriod.toMillis)
+        for {
+          roles <- DBIO
+            .from(samDAO.listUserRolesForResource(SamResourceTypeNames.workspace, workspace.workspaceId, ctx))
+          petEmail <- DBIO.from(samDAO.getUserPetServiceAccount(ctx, workspace.googleProjectId))
+          userAndPet = UserAndPetEmails(WorkbenchEmail(ctx.userInfo.userEmail.value), petEmail)
+          _ <- setupProjectRoles(workspace, roles, userAndPet, expirationDate)
+          _ <- setupBucketRoles(workspace, roles, userAndPet, expirationDate)
+          _ <- DBIO
+            .from(openTelemetry.incrementCounter("fastpass-granted-user", tags = openTelemetryTags).unsafeToFuture())
+        } yield ()
+      } else {
+        logger.info(
+          s"Not enough IAM Policy Role Binding quota available to add FastPass access for ${ctx.userInfo.userEmail.value} in workspace ${workspace.toWorkspaceName}"
+        )
+        DBIO.successful()
+      }
+    }
+  }
+
+  def setupFastPassForUserInClonedWorkspace(parentWorkspace: Workspace,
```

Comment:
```suggestion
  def setupFastPassForUserInClonedParentWorkspace(parentWorkspace: Workspace,
```
This may be kind of a nit, but upon my first read through of this, I was expecting this method to set up ALL of the FastPass permissions for the Parent AND the Child.  So the naming change is an attempt to clarify what this method does vs. `setupFastPassForUserInNewWorkspace`.  This might also benefit from some comments explaining that this method is meant to work in conjunction with `setupFastPassForUserInNewWorkspace` to setup the right permissions on the Parent from the Child. 

EDIT: A-ha!  See Christine's comment, she was also confused by this.  Add some comments to help differentiate.

Reply:
I agree that this name change is helpful… I also wasn't getting that this was for the parent workspace and not the newly created one.